src/coreComponents/mesh/ElementRegionManager.cpp

```diff
@@ -119,12 +120,26 @@ void ElementRegionManager::setSchemaDeviations( xmlWrapper::xmlNode schemaRoot,
   }
 }
 
+
 void ElementRegionManager::generateMesh( CellBlockManagerABC const & cellBlockManager )
 {
-  this->forElementRegions< CellElementRegion >( [&]( CellElementRegion & elemRegion )
-  {
-    elemRegion.generateMesh( cellBlockManager.getCellBlocks() );
-  } );
+  { // cellBlocks loading
+    Group const & cellBlocks = cellBlockManager.getCellBlocks();
+    CellElementRegionSelector cellBlockSelector{ cellBlocks };
+    this->forElementRegions< CellElementRegion >( [&]( CellElementRegion & elemRegion )
+    {
+      std::set< string > selectedCellBlocks = cellBlockSelector.buildRegionCellBlocksSelection( elemRegion );
+      elemRegion.clearCellBlockNames();
```

Comment:
For info, why do you need to clean? It may already contain something?

Reply:
Indeed, the goal here were to clear all lists (`m_cellBlockAttributeValues`, `m_cellBlockMatchPatterns`, `m_cellBlockNames`) to take into account the selection built here.
I renamed this method to `clearCellBlockInputs()`, that may be clearer.

Reply:
OK, that was not obvious to me. Could you add a little comment?